cpp/fury/util/string_util.cc

```diff
@@ -21,43 +21,20 @@
 
 #if defined(__x86_64__) || defined(_M_X64)
 #include <immintrin.h>
+#define USE_IMMINTRIN_SIMD
```

Comment:
How about adding a `platform.h` and define macro in that file. 

And I perfer naming the macro as `FURY_HAS_SSE2/FURY_HAS_NEON/...`